Glossary

thiagooak opened the pull request:
@steveklabnik Trying out the Glossary idea.

Added the paragraph about 'complicated words' because I think it would be useful to those contributing to the book. Maybe this should not be here

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @steveklabnik (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTING.md](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

steveklabnik commented on src/doc/trpl/glossary.md:
```diff
@@ -0,0 +1,17 @@
+
+In this section we briefly explain terms that you may not be familiar with.
+
+This book avoids, as much as possible, overcomplicated terms. Nevertheless, the following were deemed important enough that omitting them would be a disservice to the reader.
```
I would prefer to be a bit more positive here. I'm not sure exactly what I'd want to write, but something more like "not everyone who comes to Rust has a background in systems programming, nor in computer science, so we've added explanations of domain terms that might be unfamilliar" or something

thiagooak commented:
Really like the way you wrote it, did small changes

"Not every Rustacean has a background in systems programming, nor in computer
science, so we've added explanations of terms that might be unfamiliar."

* not sure if Rustacean is really a thing, but if it is I like the idea to add it
* maybe could be "not every *new* Rustacean" but sounds a bit demeaning
* removed domain from "domain terms" - simpler and has similar effect

steveklabnik commented:
I already mention Rustaceans, so it's fine to add here :)

steveklabnik commented:
@bors: r+ 9783cc2 rollup

steveklabnik commented:
Awesome, thanks!

kmcallister commented:
Nice! Is there a wiki page etc. where we can add terms to be glossarized?

Also, it'd be great to enhance rustdoc such that we can cross-link possibly unfamiliar terms between the glossary and the text.

steveklabnik commented:
> Is there a wiki page etc. where we can add terms to be glossarized?

Just submit pull requests, though I don't want this to get out of hand. Primarily focusing on CS terms here.
